### ISSUE TITLE
Add caching of tags

### ISSUE THREAD

**Issue by jszwedko** [jszwedko/ec2-metadatafs#1]
With a configurable cache TTL so that we don't hammer the AWS API as much.

**Comment by joshuadfranklin** [comment]
I'd also say cache the 169.254.169.254 for a bit too, when I run ec2-metadatafs with DEBUG in foreground  and autocomplete like /tmp/aws/metTAB/instTAB there are a lot of HTTP requests.

**Comment by jszwedko** [comment]
Agreed, I'll open a separate issue for caching the metadata service.
